momento-sdk/src/main/java/momento/sdk/SubscriptionWrapper.java

```diff
@@ -19,42 +21,54 @@
 public class SubscriptionWrapper implements Closeable {
   private final Logger logger = LoggerFactory.getLogger(SubscriptionWrapper.class);
   private final ScsTopicGrpcStubsManager grpcManager;
-  private final String cacheName;
-  private final String topicName;
-  private final SubscriptionState subscriptionState;
-  private final ISubscriptionCallbacks options;
-  private StreamObserver<_SubscriptionItem> subscription;
-
-  public SubscriptionWrapper(
-      ScsTopicGrpcStubsManager grpcManager,
-      String cacheName,
-      String topicName,
-      SubscriptionState subscriptionState,
-      ISubscriptionCallbacks options) {
+  private final SendSubscribeOptions options;
+  private ScheduledExecutorService scheduler;
+
+  private CancelableClientCallStreamObserver<_SubscriptionItem> subscription;
+
+  public SubscriptionWrapper(ScsTopicGrpcStubsManager grpcManager, SendSubscribeOptions options) {
     this.grpcManager = grpcManager;
-    this.cacheName = cacheName;
-    this.topicName = topicName;
-    this.subscriptionState = subscriptionState;
     this.options = options;
   }
 
-  public CompletableFuture<Void> subscribe() {
+  public CompletableFuture<Void> subscribeWithRetry() {
```

Comment:
this should be able to be fully `private` i believe

Reply:
private within the package since it is called by the ScsTopicClient class (in the same package though)